From 7a6b21f6171d048b89bd4f1317e542cd41066090 Mon Sep 17 00:00:00 2001
From: Xiao Chen <xiao@cloudera.com>
Date: Fri, 5 Feb 2016 15:17:06 -0800
Subject: [PATCH 1182/2748] CLOUDERA-BUILD. Addendum of HDFS-8576 to fix test
 failure.

Change-Id: Ib593e7fcba94a0b4951ee9c04c9e4d199c618763
---
 .../org/apache/hadoop/hdfs/TestLeaseRecovery.java  |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/TestLeaseRecovery.java b/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/TestLeaseRecovery.java
index 9ec1238..29e6e21 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/TestLeaseRecovery.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/TestLeaseRecovery.java
@@ -27,6 +27,7 @@
 import org.apache.hadoop.fs.FileSystem;
 import org.apache.hadoop.fs.Path;
 import org.apache.hadoop.hdfs.MiniDFSCluster.DataNodeProperties;
+import org.apache.hadoop.hdfs.protocol.AlreadyBeingCreatedException;
 import org.apache.hadoop.hdfs.protocol.Block;
 import org.apache.hadoop.hdfs.protocol.DatanodeInfo;
 import org.apache.hadoop.hdfs.protocol.ExtendedBlock;
@@ -249,7 +250,8 @@ public void testLeaseRecoveryAndAppend() throws Exception {
         newdfs.append(file);
         fail("Append to a file(lease is held by another client) should fail");
     } catch (RemoteException e) {
-      assertTrue(e.getMessage().contains("file lease is currently owned"));
+      Exception inner = e.unwrapRemoteException();
+      assertTrue("Exception type is wrong! " + inner, inner instanceof AlreadyBeingCreatedException);
     }
 
     // Lease recovery on first try should be successful
-- 
1.7.9.5

